### PR TITLE
Im2col

### PR DESCRIPTION
**Pull request by nikmo33**
Adds im2col for sampling

### CODE REVIEW

File: nerfstudio/data/pixel_samplers.py [hunk 1]
```diff
@@ -37,26 +37,34 @@ def collate_image_dataset_batch(batch: Dict, num_rays_per_batch: int, keep_full_
...
 
+
     c, y, x = (i.flatten() for i in torch.split(indices, 1, dim=-1))
-    image = batch["image"][c, y, x]
+    col_img = batch["image"].unfold(dimension=1, size=patch_size, step=1).unfold(dimension=2, size=patch_size, step=1)
+
+    image = col_img[c, y, x].permute(0, 2, 3, 1).reshape(-1, 3)
+    # image = batch["image"][c, y, x]
     mask, semantics_stuff, semantics_thing = None, None, None
+    collated
```
Comment (zmurez) [inline nerfstudio/data/pixel_samplers.py:60]: ?

File: nerfstudio/data/pixel_samplers.py [hunk 2]
```diff
@@ -37,26 +37,34 @@ def collate_image_dataset_batch(batch: Dict, num_rays_per_batch: int, keep_full_
     num_images, image_height, image_width, _ = batch["image"].shape
 
     # only sample within the mask, if the mask is in the batch
+    mask = torch.ones_like(batch["image"][..., 0]).bool()
     if "mask" in batch:
         nonzero_indices = torch.nonzero(batch["mask"][..., 0], as_tuple=False)
         chosen_indices = random.sample(range(len(nonzero_indices)), k=num_rays_per_batch)
         indices = nonzero_indices[chosen_indices]
     else:
```
Comment (zmurez) [inline nerfstudio/data/pixel_samplers.py:45]: I guess we only support patches for images without masks for now?

File: nerfstudio/models/nerfacto.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 import numpy as np
 import torch
 from torch.nn import Parameter
-from torchmetrics import PeakSignalNoiseRatio
+from torchmetrics import PeakSignalNoiseRatio, StructuralSimilarityIndexMeasure
 from torchmetrics.functional import structural_similarity_index_measure
```
Comment (zmurez) [inline nerfstudio/models/nerfacto.py:28]: delete since using module instead

File: nerfstudio/models/nerfacto.py [hunk 4]
```diff
@@ -243,6 +243,11 @@ def get_loss_dict(self, outputs, batch, metrics_dict=None):
         loss_dict = {}
         image = batch["image"].to(self.device)
         loss_dict["rgb_loss"] = self.rgb_loss(image, outputs["rgb"])
+        if batch["patch_size"] > 11:
```
Comment (zmurez) [inline nerfstudio/models/nerfacto.py:246]: Probably want this if to be based on the loss weights being non-zero and add loss weights as part of the config.

**Review by zmurez — COMMENTED** [review]
Seems like a reasonable start but we probably need to support patches with masks? Also a bit more configurable. Is this something we want to contribute back to nerfstudio? If so might need to branch off their main instead of ours?
